ts/components/wallet/PaymentBannerComponent.tsx

```diff
@@ -5,52 +5,100 @@
   paymentReason: string;
-  currentAmount: string;
-  entity: string;
+  currentAmount: AmountInEuroCents;
+  recipient: EnteBeneficiario;
 }>;
 
-export default class PaymentBannerComponent extends React.Component<Props> {
+type ReduxMappedDispatchProps = Readonly<{
+  showPaymentSummary: () => void;
+}>;
+
+type OwnProps = Readonly<{
+  navigation: NavigationScreenProp<NavigationState>;
+}>;
+
+type Props = OwnProps & ReduxMappedStateProps & ReduxMappedDispatchProps;
+
+class PaymentBannerComponent extends React.Component<Props> {
   public render(): React.ReactNode {
     return (
-      <Grid style={[WalletStyles.topContainer, WalletStyles.paddedLR]}>
-        <Row>
-          <Col>
-            <View spacer={true} />
-            <Text bold={true} style={WalletStyles.white}>
-              {this.props.paymentReason}
-            </Text>
-          </Col>
-          <Col>
-            <View spacer={true} />
-            <Text
-              bold={true}
-              style={[WalletStyles.white, WalletStyles.textRight]}
-            >
-              {`${this.props.currentAmount} €`}
-            </Text>
-          </Col>
-        </Row>
-        <Row>
-          <Col>
-            <Text style={WalletStyles.white}>{this.props.entity}</Text>
-            <View spacer={true} />
-          </Col>
-          <Col>
-            <Text style={[WalletStyles.white, WalletStyles.textRight]}>
-              {I18n.t("wallet.cancel").toUpperCase()}
-            </Text>
-            <View spacer={true} />
-          </Col>
-        </Row>
-      </Grid>
+      <TouchableOpacity onPress={() => this.props.showPaymentSummary()}>
+        <Grid style={[WalletStyles.topContainer, WalletStyles.paddedLR]}>
+          <Row>
+            <Col>
+              <View spacer={true} />
+              <Text bold={true} style={WalletStyles.white}>
+                {this.props.paymentReason}
+              </Text>
+            </Col>
+            <Col>
+              <View spacer={true} />
+              <Text
+                bold={true}
+                style={[WalletStyles.white, WalletStyles.textRight]}
+              >
+                {`${AmountInEuroCentsFromNumber.encode(
+                  this.props.currentAmount
+                )} €`}
+              </Text>
+            </Col>
+          </Row>
+          <Row>
+            <Col>
+              <Text style={WalletStyles.white}>
+                {this.props.recipient.denominazioneBeneficiario}
+              </Text>
+              <View spacer={true} />
+            </Col>
+            <Col>
+              <Text style={[WalletStyles.white, WalletStyles.textRight]}>
+                {I18n.t("wallet.cancel").toUpperCase()}
+              </Text>
+              <View spacer={true} />
+            </Col>
+          </Row>
+        </Grid>
+      </TouchableOpacity>
     );
   }
 }
+
+const mapStateToProps = (state: GlobalState): ReduxMappedStateProps => ({
+  paymentReason: paymentReasonSelector(state).getOrElse(""),
```

Comment:
don't hardcode default values here, instead define two constants at the top of this module: `DEFAULT_PAYMENT_REASON`, `DEFAULT_PAYMENT_AMOUNT` and `DEFAULT_PAYMENT_RECIPIENT`